File: March 2021/05.03.2021/Kaan.java
class Solution {
public:
    
    void traverse(TreeNode* root, int level,vector<vector<int>>& list){
        
        if(root == nullptr){
            return;
        }
        
        if(list.size()<level){
            vector<int> newList;
            newList.push_back(root->val);
            list.push_back(newList);
        }
        else{
            list[level-1].push_back(root->val);
        }
        traverse(root->left,level+1,list);
        traverse(root->right,level+1,list);
    }
    
    vector<double> averageOfLevels(TreeNode* root) {
        
        vector<vector<int>> list;
        vector<double> result;
        traverse(root,1,list);
        
        for(int i = 0 ; i<list.size();i++){
            
            double sum = 0;
            for(int j = 0 ;j<list[i].size();j++){
                sum+=list[i][j];
            }
            result.push_back(sum/list[i].size());
            
        }
        
        return result;
        
        
        
    }
};

File: March 2021/09.03.2021/Kaan.cpp
class Solution {
public:
    
    void traverse(TreeNode* root,int v ,int d,int currentDepth){
        
        if(root == nullptr){
            return;
        }
        else if(currentDepth == d-1){
            TreeNode* leftNode = new TreeNode(v);
            TreeNode* rightNode = new TreeNode(v);
            leftNode->left = root->left;
            rightNode->right = root->right;
            root->left = leftNode;
            root->right = rightNode;
        }
        else{
            traverse(root->left,v,d,currentDepth+1);
            traverse(root->right,v,d,currentDepth+1);
        }
        
        
        
    }
    
    TreeNode* addOneRow(TreeNode* root, int v, int d) {
        
        if(d == 1){
            TreeNode* newRoot = new TreeNode(v);
            newRoot->left = root;
            return newRoot;
        }
        traverse(root,v,d,1);
        return root;
        
    }
};
